FMSgridtools/make_hgrid/hgridobj.py

```diff
@@ -0,0 +1,288 @@
+                    attrs=dict(
+                        units="meters", 
+                        standard_name="grid_edge_y_distance",
+                    )
+                )
+
+                if out_halo > 0:
+                    dy.attrs["_FillValue"] = -9999.
+
+                var_dict['dy'] = dy
+
+            if self.angle_dx is not None:   
+                angle_dx = xr.DataArray(
+                    data=self.angle_dx[:(nyp*nxp)].reshape((nyp, nxp)),
+                    dims=["nyp", "nxp"],
+                    attrs=dict(
+                        units="degrees_east",
+                        standard_name="grid_vertex_x_angle_WRT_geographic_east",
+                    )
+                )
+
+                if out_halo > 0:
+                    angle_dx.attrs["_FillValue"] = -9999.
+
+                var_dict['angle_dx'] = angle_dx
+                    
+            if conformal != "true":
+                if self.angle_dy is not None:
+                    angle_dy = xr.DataArray(
+                        data=self.angle_dy.reshape((nyp, nxp)),
+                        dims=["nyp", "nxp"],
+                        attrs=dict(
+                            units="degrees_north",
+                            standard_name="grid_vertex_y_angle_WRT_geographic_north",
+                        )
+                    )
+
+                    if out_halo > 0:
+                        angle_dy.attrs["_FillValue"] = -9999.
+
+                    var_dict['angle_dy'] = angle_dy
+
+        if self.area is not None:
+            area = xr.DataArray(
+                data=self.area[:(ny*nx)].reshape((ny, nx)),
+                dims=["ny", "nx"],
+                attrs=dict(
+                    units="m2",
+                    standard_name="grid_cell_area",
+                )
+            )
+
+            if out_halo > 0:
+                area.attrs["_FillValue"] = -9999.
+
+            var_dict['area'] = area
+
+        if north_pole_arcx == "none":
+            arcx = xr.DataArray(
+                [self.arcx],
+                attrs=dict(
+                    standard_name="grid_edge_x_arc_type",
+                )
+            )
+        else:
+            arcx = xr.DataArray(
+                [self.arcx],
+                attrs=dict(
+                    standard_name="grid_edge_x_arc_type",
+                    north_pole=north_pole_arcx,
+                )
+            )
+
+        var_dict['arcx'] = arcx   
+
+        dataset = xr.Dataset(
+            data_vars=var_dict
+        )
+        dataset.attrs = global_attrs
+        self.dataset = dataset
+        dataset.to_netcdf(outfile)
+
+    def make_gridobj(self) -> "GridObj":
+        if self.dataset is None:
+            tile = None
+            x = None
+            y = None
+            dx = None
+            dy = None
+            angle_dx = None
+            angle_dy = None
+            area = None
+            arcx = None
+            if self.tile is not None:
```

Comment:
some of these code is repetitive from the write_out_grid function?  Can this be restructured/refactored?